### ISSUE TITLE
versions 2020 and newer on Win 7 - frequent drops to 0 up/download speed that last for several seconds

### ISSUE THREAD

**Issue by felixlang** [mullvad/mullvadvpn-app#1511]
I've encountered this problem on the stable Mullvad releases of 2020.1 and 2020.2. The problem is that, while I am connected to a Mullvad VPN server, my internet speed will very frequently (at least a few times per hour) be "clamped" down to 0 or near-0 speeds, and it will stay there for several seconds before recovering. There is no actual disconnection from Mullvad, nor are there any warning messages.

I can see the results of this both with my torrent client (qBittorrent 4.2.1 64-bit) where the up and download speeds will drop to zero, and also while playing online games where there will be instances of severe lag.

I have since reverted back to Mullvad version 2019.10, where this problem does not occur at all.

My PC is using Windows 7 Professional 64-bit with all quality and security updates installed.

**Comment by faern** [comment]
With 2020.1 we upgraded the TAP adapter driver for the virtual tunnel interface from NDIS5 to NDIS6. This is supposed to be a good thing for most users. But we are aware that it works less stable for a subset of Windows 7 users.

Since we can't programmatically determine for which users which one will work best, and NDIS5 has not been updated since ~2013, we have to opt for the newer alternative for everyone. We are not aware of any other VPN client still using NDIS5.

You can try using WireGuard and see if that is more stable? It uses another virtual network interface driver, which might be more stable for you? You can activate WireGuard under settings -> advanced in the app.

Other than trying WireGuard, there is sadly not very much you can do about the stability from what I know. Our recommendation is to upgrade to a newer version of Windows. Windows 7 did officially became unsupported on January 14 of this year. We can't keep maintaining software for an operating system that the vendor has discontinued.

**Comment by felixlang** [comment]
Thank you for replying. Since data privacy is a high level of importance for me, the prospect of switching to Windows 10 is not really an option, unless you are aware of any known reliable method of restricting Microsoft's data collection for that OS to a level that is equal to or better than that of prior versions of Windows. I am aware of a software called Blackbird that claims to do this, but there are no third-party reviews that report on whether it is either effective or even trustworthy.

I will try WireGuard to see if that makes a difference and if not, then will likely seek out other possibilities.

**Comment by faern** [comment]
Staying on an operating system that won't receive security updates won't really protect your data privacy. Using outdated software is never a good thing. You either trust your **vendor** and use their latest stuff, or you switch vendor. That's my only recommendation.

**Comment by felixlang** [comment]
Well, if nothing else, I don't have the money for buying Windows 10, nor do I have time to read and attempt to understand Microsoft's 200-page privacy policy. So switching to that OS is really not a possibility for me.

**Comment by Atrate** [comment]
Check out Windows 10 LTSC. It's lighter and much less invasive, if configured properly. reddit.com/r/piracy can help you with acquiring that.

Also, if you don't use any Windows-only apps, you could try Linux.

**Comment by faern** [comment]
It would be very helpful for us to know if WireGuard works better for you or not.

We know some users are having this kind of stability problems, but we can't reproduce them ourselves. So it would be helpful to know if recommending those users to switch over to WireGuard is a viable solution or not.

**Comment by dkwarr87** [comment]
@faern Just wanted to let you know that I am using Windows 7 64bit with Mullvad 2020.3 and I was experiencing the same issues as felix when using OVPN. Switched over to WireGuard and it's been fine since, getting full speeds without any connection drops.

Only issue is every time it connects using WireGuard it registers as a new network with a new name and you get the popup asking if its a Home, Office, or Public network. I'm up to "wg-mullvad 25" now.

**Comment by faern** [comment]
@dkwarr87 That you get new network adapters on every connect is very odd. We can't reproduce that on our Windows 7 machines. Have you tried vanilla WireGuard? If so, does it behave the same way?

**Comment by saberblaze** [comment]
@faern I would like to state that I just recently tried wireguard in Windows 7 and can confirm that it creates a new network every time it connects to a new server. I have not tried official/alternate clients.

**Comment by faern** [comment]
We have released [`2020.4-beta2`](https://github.com/mullvad/mullvadvpn-app/releases/tag/2020.4-beta2) which uses the older NDIS5 driver again on Windows 7 systems. Try upgrading to this beta and see if OpenVPN performs better.

**Comment by Evernow** [comment]
Supporting Windows 7, an OS that is not even supported by its makers anymore is just pointless. 

There's no reason to be on that operating system, if one has an application that cannot run on it then virtual machines exist, in that case the host would be a properly supported operating system.

**Comment by faern** [comment]
We indeed indeed strongly recommend that you do not run software that does no longer receive security patches (https://github.com/mullvad/mullvadvpn-app/issues/1511#issuecomment-588495779). So we strongly advise against running Windows 7. But at the same time we know we have a bunch of users who do use it. So we will support it for a while longer.

However, we will also stop supporting it in a not too distant future. It's a 10 year old OS and software does not live forever/age well. So we strongly recommend all our Windows 7 users to change to a supported operating system asap.

I'm closing this issue since the speed issues and packet drops should have been resolved in the latest beta.
